diff --git a/CMSIS/Include/cmsis_gcc.h b/CMSIS/Include/cmsis_gcc.h
index d868f2e..808ea7d 100644
--- a/CMSIS/Include/cmsis_gcc.h
+++ b/CMSIS/Include/cmsis_gcc.h
@@ -337,7 +337,11 @@ __attribute__( ( always_inline ) ) __STATIC_INLINE void __set_FPSCR(uint32_t fps
 #if (__FPU_PRESENT == 1U) && (__FPU_USED == 1U)
   /* Empty asm statement works as a scheduling barrier */
   __ASM volatile ("");
+#if defined (__clang__)
+  __ASM volatile ("VMSR fpscr, %0" : : "r" (fpscr) :);
+#else
   __ASM volatile ("VMSR fpscr, %0" : : "r" (fpscr) : "vfpcc");
+#endif
   __ASM volatile ("");
 #endif
 }
